### DISCUSSION TITLE
Use Prisma DB connector

### DISCUSSION THREAD

**Discussion by samchouse** [rwf2/Rocket#2199] [Questions] [answered]
I would like to use Prisma as my ORM but I don't know how to use it with Rocket. I know I can probably manage it using Rocket but as far as I've read pooling is better so I'm trying to do that. I'm using https://github.com/Brendonovich/prisma-client-rust as the Prisma client. I know I have to implement some custom logic for this to work with Rocket but I'm not sure what it is. How would I get this working?

**Comment by the10thWiz** [comment, marked as answer]
It looks like you would need to implement the pool logic to make this work. This would likely mean implementing `Pool` from `rocket_db_pools`, but there are a few caveats. First, all of the supported db connectors take advantage of existing pool implementations (e.g., Rocket uses `sqlx::Pool` to handle pooling database connections), but I didn't see a pool implementation for prisma-client.

Right now, I don't see a good way to use prisma-client with pooling. You could open an issue on prisma-client to add pooling support (or maybe there's something I'm not aware of making it unnessecary), but the general pattern would be to build the pool, and add it as managed state to Rocket.
